ql/src/java/org/apache/hadoop/hive/ql/stats/StatsUtils.java

```diff
@@ -647,7 +647,7 @@ private static Range getRangePartitionColumn(PartitionIterable partitions, Strin
       double max = Double.MIN_VALUE;
       for (Partition partition : partitions) {
         partVal = partition.getSpec().get(partColName);
-        if (!partVal.equals(defaultPartName)) {
+        if (NumberUtils.isCreatable(partVal)) {
```

Comment:
can we merge `DECIMAL_TYPE` with `DOUBLE_TYPE` handling ? The use of BigDecimal followed by a cast to double seems unnecessary.

Reply:
thanks!